geth/jail/jail.go

```diff
@@ -43,6 +48,85 @@ type Jail struct {
 	baseJSCode     string                     // JavaScript used to initialize all new cells with
 }
 
+// Copy returns a new JailCell instance with a new eventloop runtime associated with
+// the given cell.
+func (cell *JailCell) Copy() (common.JailCell, error) {
+	vmCopy := cell.vm.Copy()
+	loCopy := loop.New(vmCopy)
+
+	// Register fetch provider from ottoext.
+	if err := fetch.Define(vmCopy, loCopy); err != nil {
+		return nil, err
+	}
+
+	// Register event loop for timers.
+	if err := timers.Define(vmCopy, loCopy); err != nil {
+		return nil, err
+	}
+
+	return &JailCell{
+		id:  cell.id,
+		vm:  vmCopy,
+		lo:  loCopy,
+		sem: semaphore.New(1, JailCellRequestTimeout*time.Second),
+	}, nil
+}
+
+// Fetch attempts to call the underline Fetch API added through the
+// ottoext package.
+func (cell *JailCell) Fetch(url string, callback func(otto.Value)) (otto.Value, error) {
+	if err := cell.vm.Set("__captureFetch", func(res otto.Value) {
```

Comment:
Sorry, let me elaborate. Initially I wanted to remove the need for dealing directly with the `otto.Value` primitive, but faced some issues doing so, because the internal `js` was calling `JSON.stringify`, which would just block. Am yet to figure out why. 

I will make the changes as reviewed, but we should keep in mind that sooner or later we will need to deal with the underlying issue of why in js atleast `JSON.stringify` would just block in an otto vm context.